js/app/courseHistoryStore.js

```diff
@@ -44,11 +44,13 @@ var CourseHistoryStore = new GObject.Class({
                         } else {
                             this.set_current_subset(payload.model);
                         }
-                    } else {
+                    } else if (payload.model instanceof Eknc.MediaObjectModel) {
                         this.set_current_item_from_props({
                             media_model: payload.model,
                             context: payload.context,
                         }, EntryPoints.LINK_CLICKED);
+                    } else {
+                        this.show_ekn_id(payload.model.ekn_id);
```

Comment:
Not sure I understand, can you elaborate?

What I want to achieve is that, when an text-based article object (non-media and non-set) is clicked, whether is a  ITEM_CLICKED (from GTK UI) or ARTICLE_LINK_CLICKED (from webkit) it should always open it in the article page, otherwise just open it in the light-box as usual.

Reply:
I think I get it now..